net-im/profanity/profanity-0.14.0-r1.ebuild

```diff
@@ -14,6 +14,8 @@ SRC_URI="
         -> ${PN}-0.14.0-ox-carbons.patch
     https://github.com/profanity-im/profanity/commit/2ed6211c.patch
         -> ${PN}-0.14.0-xscreensaver.patch
+    https://github.com/profanity-im/profanity/commit/b8817470.patch
```

Comment:
I'm really not keen on downloading patches like this, but I know Flow asked you to use them, so I won't block on it right now.

I don't think there's any guarantee that the links are stable and we've had them change in the past for PRs at least.

It's a cheap hack which doesn't really gain us much at all and just has downsides.

Reply:
Yeah it's a good point that its safer to have them locally.
At least in in this case I can guarantee that the commit will not vanish since I will not force push to the upstream repo and the commits are not from branches/PRs. So as long as GH is there the patch (and source tarball) should be as wel.

Reply:
Thanks! https://github.com/orgs/community/discussions/56522 is part of why I get nervous over these too btw.

Reply:
@thesamesam Patch URLs that reference a commit are by definition stable of their content (minus using short commit IDs as done here, which could collide), hence the content is guranteed to be stable.

What you may have in mind, besides the discussion regarding diff patches, are PR Patch URLs like https://github.com/gentoo/gentoo/pull/32792.patch. Those can change, as the PR contents change.

I find Commit Patch URLs in SRC_URI very convenient. They are guranteed to be stable, do not consume space and `FILESDIR`, and therefore, are not prone to cause unused files in `FILESDIR`.

Reply:
I'm not sure they are guaranteed to be stable (github has a history of instability even with things we expect to be reliable, like their generated tarballs), but in any case, this is the kind of thing where I don't really feel you should _ask_ proxied maints to do it.

Feel free to do it for your own ebuilds, but if we're going to ask proxied maints to do it, we need to discuss it within the project for consistency, otherwise we just have a bunch of ebuilds doing different things depending on who the reviewer was. There's nothing intrinsically better about them, really, so it's not like it's obvious they should do it.

Reply:
disregarding merge commits, a regular commit implemented as a .patch by a non-attacker with trusted contents that you vetted at the time of reviewing the ebuild update...

... can still change over time. It's not about faked malicious contents. It's about the fact that unlike `git-archive` (which is generally stable, had one issue with changing the default compressor, and GitHub has acknowledged this as a bug and said they intend to offer byte-for-byte reproducibility of archive/refs/tags/* tarballs),

the contents of a .patch url are not stable. They are equivalent to running `git format-patch` on the commit in question, but with the trailing footer containing the git version number removed, otherwise .patch urls on github would change ***all the time***. However, this is not the only source of instability. Stuff like this:

```
index 6be74bedc..b58e9290b 100644
```

can change over time as the git repository becomes orders of magnitude bigger, and git, running on the underlying backend repository, calculates that it needs to start truncating the hashes to larger lengths in order to remain unique.

The same issue manifests for example with `git describe`. The general solution here is to pass `--abbrev=40` as a preventative measure, since you don't really know how fast your repository will grow and how long old .patch urls will remain interesting to users.

There's also a general issue of the diffstat not necessarily being guaranteed to use the same format in newer git versions, but I don't recall offhand specific examples of this being an issue.

Reply:
All that being said, it's not particularly common for this to change. It's "usually" pretty stable, and the times when it becomes unstable are when a repository grows in size by an order of magnitude.